cpp/ql/lib/semmle/code/cpp/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -0,0 +1,245 @@
+    // needed to support `Argument[x..y]` ranges and `Argument[-1]`
+    token.getName() = "Argument" and
+    exists(int pos | pos = AccessPath::parseInt(token.getAnArgument()) |
+      result = TDirectPosition(pos)
+    )
+  }
+
+  bindingset[token]
+  ArgumentPosition decodeUnknownArgumentPosition(AccessPath::AccessPathTokenBase token) {
+    // needed to support `Parameter[x..y]` ranges and `Parameter[-1]`
+    token.getName() = "Parameter" and
+    exists(int pos | pos = AccessPath::parseInt(token.getAnArgument()) |
+      result = TDirectPosition(pos)
+    )
+  }
+}
+
+/*private*/ import Make<DataFlowImplSpecific::CppDataFlow, Input> as Impl
+
+private module StepsInput implements Impl::Private::StepsInputSig {
+  DataFlowCall getACall(Public::SummarizedCallable sc) { result.getStaticCallTarget() = sc }
+}
+
+module SourceSinkInterpretationInput implements
+  Impl::Private::External::SourceSinkInterpretationInputSig<Location>
+{
+  class Element = Element;
+
+  class SourceOrSinkElement = Element;
+
+  /**
+   * Holds if an external source specification exists for `e` with output specification
+   * `output`, kind `kind`, and provenance `provenance`.
+   */
+  predicate sourceElement(SourceOrSinkElement e, string output, string kind) {
+    exists(
+      string namespace, string type, boolean subtypes, string name, string signature, string ext
+    |
+      sourceModel(namespace, type, subtypes, name, signature, ext, output, kind, _) and
+      e = interpretElement(namespace, type, subtypes, name, signature, ext)
+    )
+  }
+
+  /**
+   * Holds if an external sink specification exists for `e` with input specification
+   * `input`, kind `kind` and provenance `provenance`.
+   */
+  predicate sinkElement(SourceOrSinkElement e, string input, string kind) {
+    exists(
+      string package, string type, boolean subtypes, string name, string signature, string ext
+    |
+      sinkModel(package, type, subtypes, name, signature, ext, input, kind, _) and
+      e = interpretElement(package, type, subtypes, name, signature, ext)
+    )
+  }
+
+  private newtype TInterpretNode =
+    TElement_(Element n) or
+    TNode_(Node n)
+
+  /** An entity used to interpret a source/sink specification. */
+  class InterpretNode extends TInterpretNode {
+    /** Gets the element that this node corresponds to, if any. */
+    SourceOrSinkElement asElement() { this = TElement_(result) }
+
+    /** Gets the data-flow node that this node corresponds to, if any. */
+    Node asNode() { this = TNode_(result) }
+
+    /** Gets the call that this node corresponds to, if any. */
+    DataFlowCall asCall() { this.asElement() = result.getAst() }
+
+    /** Gets the callable that this node corresponds to, if any. */
+    DataFlowCallable asCallable() { result.(Function) = this.asElement() }
+
+    /** Gets the target of this call, if any. */
+    Element getCallTarget() { result = this.asCall().getAst().(Call).getTarget() }
```

Comment:
I think this is equivalent, but avoids going to the AST world:
```suggestion
    Element getCallTarget() { result = this.asCall().getStaticCallTarget() }
```

Reply:
Yep, seems to work just as well.